package sample_bank_account;

/**
 * @Indexed
 * @TypeId(1000042)
 */
message User {

   /**
    * @Field(index = Index.YES, store = Store.YES)
    */
   required int32 id = 1;

   /**
    * @Field(store = Store.YES)
    */
   repeated int32 accountIds = 2;

   /**
    * @Field(store = Store.YES)
    */
   required string name = 3;

   /**
    * @Field(store = Store.YES)
    */
   required string surname = 4;

message Address {
   /**
    * @Field(store = Store.YES)
    */
      required string street = 1;
   /**
    * @Field(store = Store.YES)
    */
      required string postCode = 2;
   /**
    * @Field(store = Store.YES)
    */
      required int32 number = 3;
   }


   /**
    * @Field(store = Store.YES)
    */
   repeated Address addresses = 5;     //a repeated field cannot be marked required

   /**
    * @Field(store = Store.YES)
    */
   optional int32 age = 6;   // persisting age instead of birth date is not ideal but is ok for our sample code

/**
 * @TypeId(1000043)
 */
   enum Gender {
      MALE = 0;    //NOTE: It seems negative values are accepted too!
      FEMALE = 1;
   }

   /**
    * @Field(store = Store.YES)
    */
   optional Gender gender = 7;

   optional string notes = 8;
}
